python/lsst/analysis/ap/apdb.py

```diff
@@ -116,21 +102,17 @@ def _make_flag_exclusion_query(self, query, table, flag_list, column_name='flags
             Query to include the where statement in.
         table : `sqlalchemy.schema.Table`
             Table containing the column to be queried.
-        column_name : `str`, optional
-            Name of flag column to query.
 
         Returns
         -------
-        clause : `str`
-            Clause to include in the SQL where statement.
+        query : `sqlalchemy.sql.Query`
+            Query that selects rows to exclude based on flags.
         """
-
-        bitmask = int(self._unpacker.makeFlagBitMask(flag_list, columnName=column_name))
-
-        if bitmask == 0:
-            warnings.warn(f"Flag bitmask is zero. Supplied flags: {flag_list}", RuntimeWarning)
-
-        return query.where(table.columns[column_name].op("&")(bitmask) == 0)
+        # Build a query that selects any source with one or more chosen flags,
+        # and return the opposite (`not_`) of that query.
+        query = query.where(sqlalchemy.not_(sqlalchemy.or_(table.columns[flag_col] == 1
+                                            for flag_col in flag_list)))
```

Comment:
While I have convinced myself it's equivalent to say, "exclude rows with one or more of these flags set," as it is to say, "select only rows that only have none of these flags set," the former makes more sense to me given the scientific context. If you're concerned this will slow the query down, happy to revisit later.